src/checkout_function.tf

```diff
@@ -76,3 +79,66 @@ module "checkout_function" {
 
   tags = var.tags
 }
+
+resource "azurerm_monitor_autoscale_setting" "checkout_function" {
+  count = var.checkout_enabled ? 1 : 0
+
+  name                = format("%s-%s-autoscale", local.project, module.checkout_function[0].name)
+  resource_group_name = azurerm_resource_group.checkout_be_rg[0].name
+  location            = var.location
+  target_resource_id  = module.checkout_function[0].app_service_plan_id
+
+  profile {
+    name = "default"
+
+    capacity {
+      default = var.checkout_function_autoscale_default
+      minimum = var.checkout_function_autoscale_minimum
+      maximum = var.checkout_function_autoscale_maximum
+    }
+
+    rule {
+      metric_trigger {
```

Comment:
the idea is to scale only according to the number of requests received in a period of time? if so I think it's fine, maybe with some performance tests I'm curious to understand the correlation with the CPU. (I wouldn't really take memory into account, due to the code of these API)